packages/nvidia_nat_core/src/nat/runtime/user_manager.py

```
@@ -0,0 +1,247 @@
+            payload: Discriminated union of JWT, API key, or basic auth credentials.
+
+        Returns:
+            A ``UserInfo`` with a deterministic user ID.
+
+        Raises:
+            ValueError: If the payload cannot be resolved to a valid user identity.
+        """
+        if isinstance(payload, JwtAuthPayload):
+            raw_token: str = payload.token.get_secret_value()
+            if not raw_token or raw_token.count(".") != 2:
+                raise ValueError("JWT token is empty or malformed (expected 3 dot-separated parts)")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)